src/target/riscv/riscv.c

```diff
@@ -852,24 +862,38 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 		struct trigger_request_info lt_1 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
 				fields.match.lt,
-			.tdata2 = trigger->address,
+			.tdata2 = trigger->address + trigger->length,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		struct trigger_request_info ge_2 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.disable |
 				fields.match.ge,
-			.tdata2 = trigger->address + trigger->length,
+			.tdata2 = trigger->address,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		ret = try_setup_chained_match_triggers(target, trigger, lt_1, ge_2);
 		if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
 			return ret;
 	}
 	LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
+	if (trigger->length > 1) {
```

Comment:
Please, consider displaying the debug & warn messages only if the setting of the exact-match trigger succeeds.

If `try_setup_single_match_trigger()` fails, I believe it does not make sense to display this message. In such case, there will be already a different error message printed.

```
LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
struct trigger_request_info eq = { ... };
retval = try_setup_single_match_trigger(target, trigger, eq);

if (retval == ERROR_OK && trigger->length > 1) {
	/* The exact-match trigger was successfully set up but the original 
	request was for a range-match trigger, which the target could not 
	accommodate. Warn the user about this limitation of the target. */
	LOG_TARGET_DEBUG(...);
	if (!info->range_trigger_fallback_encountered) {
		/* ... */
		LOG_TARGET_WARNING(...);
	}
	info->range_trigger_fallback_encountered = true;
}

return retval;
```

Reply:
Addressed.